### PR TITLE
UI Overhaul, Document Spec Migration

### PR DESCRIPTION
**Pull request by shueja-personal**
DRAFT until otherwise noted.

### CODE REVIEW

File: spec/DocumentSpecification.md
Comment (jlbabilino) [inline spec/DocumentSpecification.md:1]: I think we should bump up the version of the spec up to v0.0.1 with this change

File: src-tauri/Cargo.toml
Comment (jlbabilino) [inline src-tauri/Cargo.toml:1]: I think this change is supposed to not be committed, correct?

Comment (shueja-personal) [reply src-tauri/Cargo.toml:1]: I’ll revert it before merging

File: spec/DocumentSpecification.md
Comment (shueja-personal) [reply spec/DocumentSpecification.md:1]: I thought we weren’t bothering with spec versioning until release?

**Review by Autumn-Ou — APPROVED** [review]
unneeded comments removed, imports cleaned, code formatted, ts-ignore suppressed false error on context stable and fully functional from testing.

File: src/components/navbar/Navbar.tsx [hunk 1]
```diff
@@ -1,149 +1,52 @@
...
-              </IconButton>
-            </span>
-          </Tooltip>
-          <Tooltip title="Settings">
-            <span>
-              <IconButton
-                color="primary"
-                onClick={() => this.context.uiState.setPageNumber(2)}
+          {NavbarItemData.map((item, index) => (
+            <Tooltip title={item.name}>
```
Comment (shueja-personal) [inline src/components/navbar/Navbar.tsx:34]: The Tooltip component needs to have value={`${index}`}. TS will complain, but this works around a MUI bug where ToggleButtonGroup needs value props on its direct children to do highlighting of the selected option correctly.

File: src/components/field/PathSelector.tsx
Comment (shueja-personal) [inline src/components/field/PathSelector.tsx:1]: This whole file can be nuked actually, it’s obsolete since path selection is in the sidebar.

Comment (Autumn-Ou) [reply src/components/field/PathSelector.tsx:1]: no it's needed deleting it removes it from the side bar

Comment (Autumn-Ou) [reply src/components/field/PathSelector.tsx:1]: I just tried

**Review by jlbabilino — APPROVED** [review]

**Review by jlbabilino — APPROVED** [review]

**Pull request merged**